From 8a510462702be01d99631eb133bd6727322a38a1 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Mon, 10 Sep 2018 17:04:48 +0200
Subject: [PATCH 0684/1051] PCI: armada8k: stop relying on phy-names properties

The phy-names properties was not needed, the phy count can be obtained
without mentioned property.

At the occasion error handling was fixed.

Change-Id: I97f64e6c794e93f3822e392abc22a096b9cfa7da
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/pci/controller/dwc/pcie-armada8k.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/drivers/pci/controller/dwc/pcie-armada8k.c b/drivers/pci/controller/dwc/pcie-armada8k.c
index 6f0362914490..ac1d341d55f8 100644
--- a/drivers/pci/controller/dwc/pcie-armada8k.c
+++ b/drivers/pci/controller/dwc/pcie-armada8k.c
@@ -218,21 +218,22 @@ static int armada8k_phy_config(struct platform_device *pdev,
 			       struct armada8k_pcie *pcie)
 {
 	struct phy *comphy;
-	char phy_name[10];
 	int err, phy_count;
 	char i;
 
-	phy_count = of_property_count_strings(pdev->dev.of_node, "phy-names");
+	phy_count = of_count_phandle_with_args(pdev->dev.of_node, "phys",
+					       "#phy-cells");
 
 	if (phy_count <= 0)
 		return 0;
 
-	dev_err(&pdev->dev, "after phy count");
 	for (i = 0; i < phy_count; i++) {
-		snprintf(phy_name, sizeof(phy_name), "pcie-phy%d", i);
-		comphy = devm_phy_get(&pdev->dev, phy_name);
-		if (IS_ERR(comphy))
+		comphy = devm_of_phy_get_by_index(&pdev->dev,
+						  pdev->dev.of_node, i);
+		if (IS_ERR(comphy)) {
 			dev_err(&pdev->dev, "Failed to get phy %d\n", i);
+			return PTR_ERR(comphy);
+		}
 
 		switch (phy_count) {
 		case PCIE_LNK_X1:
-- 
2.17.1

